src/hotspot/share/opto/vectornode.hpp

```diff
@@ -227,47 +229,87 @@ class ReductionNode : public Node {
 
   // Needed for proper cloning.
   virtual uint size_of() const { return sizeof(*this); }
-};
 
-//---------------------------UnorderedReductionNode-------------------------------------
-// Order of reduction does not matter. Example int add. Not true for float add.
-class UnorderedReductionNode : public ReductionNode {
-public:
-  UnorderedReductionNode(Node * ctrl, Node* in1, Node* in2) : ReductionNode(ctrl, in1, in2) {
-    init_class_id(Class_UnorderedReduction);
+  // Floating-point addition and multiplication are non-associative, so
+  // AddReductionVF/D and MulReductionVF/D require strict-ordering
+  // in auto-vectorization. Currently, Vector API allows
+  // AddReductionVF/VD to be associative (no need of strict ordering)
+  // which can benefit some platforms.
+  //
+  // Other reductions are associative (do not need strict ordering).
+  virtual bool is_associative() const {
```

Comment:
I think this flag may be badly named. The idea you want to express is not so much associativity, but whether such nodes should be treated as strictly ordered. It would be much less confusing to pick a name like ordered() because that describes what you want to the node to do.

Reply:
I think you are right. Maybe we can use `ordered/unordered`. I just want to make sure we don't have too many synonyms (ordered, unordered, associative, strictly ordered, etc). But I guess associative and ordered is not exactly a synonym.

Reply:
Hi, thank you for your comments on this. I personally also feel "is_associative" is a bit non-intuitive as in the reader might have to make the connection between "associativity" and "ordering" compared to the case where we directly use what we intend the variable to do, something like "is_ordered". Would it be okay if I revert this to "requires_strict_order" for the variable and method names that I used in my first commit?

Reply:
Yes, I think that would be ok.

Reply:
Thank you. I have reverted to "requires_strict_order" in the new PS.